File: Verde.2/Palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int Verificapalindromo(char str[]){
    int inicio = 0;
    int ehpalindromo = 1;
    int fim = strlen(str) - 1;

    while(ehpalindromo == 1 && inicio < fim){
        if(str[inicio]!=str[fim]){
            ehpalindromo = 0;    //verifica se é ou não Palindromo
        }
        inicio++;    
        fim--;

}
    return ehpalindromo;

}
int main(){
    char str[500];
    char teste[] = "FIM";

    while (fgets(str, sizeof(str), stdin) != NULL) {  
        str[strcspn(str, "\n")] = '\0';     //recebe a entrada de dados e chama a função
        if(strcmp(teste, str) != 0){

        if(Verificapalindromo(str) == 1) printf("SIM\n");
        else printf("NAO\n");
    }
   }
  
return 0;
}

File: Verde.2/Palindromo.java

class Palindromo {

    public static boolean Verificapalindromo(String x) {
        int esq = 0;
        int dir = x.length() - 1;
        while (esq < dir) {
            if (x.charAt(esq) != x.charAt(dir)) {
                return false;
            }
            esq++;
            dir--;
        }
        return true;
    }

    public static boolean testeFim(String x) {
        boolean ehfim = false;
        if (x.length() == 3) {
            if (x.charAt(0) == 'F' && x.charAt(1) == 'I' && x.charAt(2) == 'M') {
                ehfim = true;
            }
        }
        return ehfim;
    }

    public static void main(String args[]) {
        String texto = MyIO.readLine();
        while (!testeFim(texto)) {
            if (Verificapalindromo(texto)) {
                MyIO.println("SIM");
            } else {
                MyIO.println("NAO");
            }
            texto = MyIO.readLine();
        }
    }
}
